test/js/specs/happy.js

```diff
@@ -2,125 +2,424 @@ const path = require('path')
+      documentSelection.identityCardHint.isDisplayed()
+    })
   })
 })
 
-  describe('upload screen', function () {
-    it('should upload a file', async () => {
+  //Document upload screen
+  describe('document upload screen', function () {
+
+    const copy = locale("en")
+    const documentUploadLocale = copy
+    const documentUploadConfirmationLocale = copy
+    const goToPassportUploadScreen = async (parameter) => {
+      if (typeof parameter === 'undefined') {
+          parameter = ''
+        }
+      driver.get(localhostUrl + parameter)
+      welcome.primaryBtn.click()
+      documentSelection.passportIcon.click()
+  }
+
+    function waitForUploadToFinish() { return (async () => {
+      const confirmBtn = await this.$('.onfido-sdk-ui-Confirm-btn-primary')
+      await driver.wait(until.elementIsVisible(confirmBtn),5000);
+    })}
+
+    it('should display cross device icon', async () => {
+      goToPassportUploadScreen()
+      documentUpload.crossDeviceIcon.isDisplayed()
+    })
+
+    it('should display cross device header', async () => {
+      const crossDeviceHeaderText = documentUpload.crossDeviceHeader.getText()
+      expect(crossDeviceHeaderText).to.equal(documentUploadLocale["cross_device"]["switch_device"]["header"])
+      documentUpload.crossDeviceHeader.isDisplayed()
+    })
+
+    it('should display cross device submessage', async () => {
+      const crossDeviceSubMessageText = documentUpload.crossDeviceSubMessage.getText()
+      expect(crossDeviceSubMessageText).to.equal(documentUploadLocale["cross_device"]["switch_device"]["submessage"])
+      documentUpload.crossDeviceSubMessage.isDisplayed()
+    })
+
+    it('should display cross device arrow', async () => {
+      documentUpload.crossDeviceArrow.isDisplayed()
+    })
+
+    it('should display uploader icon', async () => {
+      documentUpload.uploaderIcon.isDisplayed()
+    })
+
+    it('should display uploader button', async () => {
+      const uploaderBtnText = documentUpload.uploaderBtn.getText()
+      expect(uploaderBtnText).to.equal(documentUploadLocale["capture"]["upload_file"])
+      documentUpload.uploaderBtn.isDisplayed()
+    })
+
+    it('should upload a passport and verify UI elements', async () => {
+      goToPassportUploadScreen()
+      const passportTitleText =  documentUpload.title.getText()
+      expect(passportTitleText).to.equal(documentUploadLocale["capture"]["passport"]["front"]["title"])
+      documentUpload.title.isDisplayed()
+      const passportInstructionMessage = documentUpload.uploaderInstructionsMessage.getText()
+      expect(passportInstructionMessage).to.equal(documentUploadLocale["capture"]["passport"]["front"]["instructions"])
+      documentUpload.uploaderInstructionsMessage.isDisplayed()
+      const input = documentUpload.getUploadInput()
+      input.sendKeys(path.join(__dirname, documentUpload.assignDir('passport.jpg')))
```

Comment:
the sending of the file is quite an atomical action that could be inside of `documentUpload`. Something like `documentUpload.upload('passport.jpg')`, would make the code easier to read.

Reply:
Added in 1d01382